bolt-sidecar/src/common/backoff.rs

```diff
@@ -19,6 +19,24 @@ where
     Retry::spawn(backoff, fut).await
 }
 
+/// Retry a future with exponential backoff and jitter if the error matches a condition.
+pub async fn retry_with_backoff_if<F, T, E>(
```

Comment:
wait, didn't we have this already somewhere? or was it on the rpc?

Reply:
This is the `if` version, but it might be we have it already and I have to pull from upstream